packages/protocol/src/utils/StreamPartitionID.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * SPID - Stream Partition ID
+ */
+export default class SPID {
+    /** stream id */
+    public readonly id: string
+    /** stream partition */
+    public readonly partition: number
+    /** toString/fromString separator */
+    protected static readonly SEPARATOR = '|'
```

Comment:
Maybe we could use some char which is easily writable when using the command line. E.g. the CLI tool would then be able to use the same semantics. 

```
streamr stream subscribe 0x12345/foo/bar#123
```

The streamID can contain https://base64.guru/standards/base64url characters, so we can use e.g. colon or hash if we want. Maybe hash char would be great as our use case is be very similar to to URL hash anchor concept (a pointer into a document).